optuna/distributions.py

```diff
@@ -263,6 +263,12 @@ class DiscreteUniformDistribution(FloatDistribution):
         ValueError:
             If ``low`` value is larger than ``high`` value, or ``q`` value is smaller or
             equal to 0.
+
+    Attributes:
+        low:
+            Lower endpoint of the range of the distribution. ``low`` is included in the range.
+        high:
+            Upper endpoint of the range of the distribution. ``high`` is included in the range.
```

Comment:
I agree about the redundancy. Not sure though if arguments should be expected by a user to all be exposed as public attributes. I talked with @himkt about a related topic the other day, that ideally, all public attributes are listed in the list further (where `q` is defined).

Reply:
> ideally, all public attributes are listed in the list further down (where q is defined)

Yes, I think this is the best solution. A minor problem with this approach is that Sphinx seems to render `Attributes` section differently from properties docstrings (current [FrozenTrial](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.FrozenTrial.html) documentation is a good example), even though these two are the same as far as end user is concerned. Unless there is some clever Sphinx config somewhere (idk, didn't have chance to look), it would force us to define all public attributes via properties (with getters and setters) like e.g.

```python
@property
def low():
    return self._low
``` 

Which does not seem pythonic.

Reply:
I and @himkt came to more or less the exact same conclusions. Let me create an issue to discuss that.

Reply:
Tried creating https://github.com/optuna/optuna/issues/3285. 
Feel free to fill in or add comments. 🙇